pinot-core/src/test/java/org/apache/pinot/core/segment/index/loader/SegmentPreProcessorTest.java

```diff
@@ -140,6 +154,124 @@ private void constructV3Segment()
+    ColumnMetadata columnMetadata = segmentMetadata.getColumnMetadataFor(EXISTING_STRING_COL_RAW);
+    Assert.assertNotNull(columnMetadata);
+    Assert.assertFalse(columnMetadata.hasTextIndex());
+
+    try (SegmentPreProcessor processor = new SegmentPreProcessor(_indexDir, _indexLoadingConfig, _newColumnsSchemaWithText)) {
+      processor.process();
+      segmentMetadata = new SegmentMetadataImpl(_indexDir);
+      columnMetadata = segmentMetadata.getColumnMetadataFor(EXISTING_STRING_COL_RAW);
+
+      Assert.assertEquals(columnMetadata.getCardinality(), 5);
+      Assert.assertEquals(columnMetadata.getTotalDocs(), 100000);
+      Assert.assertEquals(columnMetadata.getTotalRawDocs(), 100000);
+      Assert.assertEquals(columnMetadata.getTotalAggDocs(), 0);
+      Assert.assertEquals(columnMetadata.getDataType(), FieldSpec.DataType.STRING);
+      Assert.assertEquals(columnMetadata.getBitsPerElement(), 3);
+      Assert.assertEquals(columnMetadata.getColumnMaxLength(), 0);
+      Assert.assertEquals(columnMetadata.getFieldType(), FieldSpec.FieldType.DIMENSION);
+      Assert.assertFalse(columnMetadata.isSorted());
+      Assert.assertFalse(columnMetadata.hasNulls());
+      Assert.assertFalse(columnMetadata.hasDictionary());
+      Assert.assertTrue(columnMetadata.hasTextIndex());
+      Assert.assertTrue(columnMetadata.isSingleValue());
+      Assert.assertEquals(columnMetadata.getMaxNumberOfMultiValues(), 0);
+      Assert.assertEquals(columnMetadata.getTotalNumberOfEntries(), 100000);
+      Assert.assertFalse(columnMetadata.isAutoGenerated());
+      Assert.assertEquals(columnMetadata.getDefaultNullValueString(), "null");
+
+      try (SegmentDirectory segmentDirectory = SegmentDirectory.createFromLocalFS(_indexDir, ReadMode.mmap);
+          SegmentDirectory.Reader reader = segmentDirectory.createReader()) {
+        Assert.assertTrue(reader.hasIndexFor(EXISTING_STRING_COL_RAW, ColumnIndexType.TEXT_INDEX));
+      }
+    }
+  }
+
+  /**
+   * Test to check for failure case where text index is enabled on an existing
+   * column that is dictionary encoded. This is currently not supported.
+   * @throws Exception
+   */
+  @Test
+  public void testEnableTextIndexOnExistingColumnDictEncoded() throws Exception {
+    constructV3Segment();
+    Set<String> textIndexColumns = new HashSet<>();
+    textIndexColumns.add(EXISTING_STRING_COL_DICT);
+    _indexLoadingConfig.setTextIndexColumns(textIndexColumns);
+    SegmentMetadataImpl segmentMetadata = new SegmentMetadataImpl(_indexDir);
+    ColumnMetadata columnMetadata = segmentMetadata.getColumnMetadataFor(EXISTING_STRING_COL_DICT);
+    Assert.assertNotNull(columnMetadata);
+    Assert.assertFalse(columnMetadata.hasTextIndex());
+
+    try (SegmentPreProcessor processor = new SegmentPreProcessor(_indexDir, _indexLoadingConfig, _newColumnsSchemaWithText)) {
+      processor.process();
+      Assert.fail("operation should have failed");
+    } catch (Exception e) {
```

Comment:
It is best to check for as close to the exception thrown as possible. Otherwise, if we throw some exception from some other part of the code, we will catch that here and assume that the test passed.
I would not be opposed to checking the exact exception message here, with the column name, for example. It will also help us refine our messages to be consistent when the exceptions are thrown

Reply:
Assert.fail ensures that if exception is not thrown by line above, the test will fail. I am now checking for the full error message including column name

Reply:
Assert.fail ensures that if exception is not thrown by line above, the test will fail. I am now checking for the full error message including column name